### DISCUSSION TITLE
Problems with implementing custom module

### DISCUSSION THREAD

**Discussion by olena112** [omnetpp/omnetpp#1266] [Q&A]
Hi everyone. I have only been using omnet++ for a month. I was trying to create a custom module for mac. I created all the necessary files for this, but when I write the simulation, it shows several errors that I can't fix. I tried to fix it in various ways that were advised. The project includes inet, also according to the advice, I added it to preferences --> compile --> included paths... 
Can someone explain how to fix these errors (I will attach below)? I can send the files if needed

```
make MODE=release all 
cd src && /usr/bin/make
make[1]: Entering directory '/c/Users/*/Downloads/omnetpp-6.1-windows-x86_64/omnetpp-6.1/samples/DcwTest/src'
DCWMac.cc
In file included from DCWMac.cc:1:
**./DCWMac.h:5:10: fatal error: 'inet/linklayer/base/MacProtocolBase.h' file not found**
#include "inet/linklayer/base/MacProtocolBase.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
**make[1]: *** [Makefile:114: ../out/clang-release/src/DCWMac.o] Error 1
make[1]: Leaving directory '/c/Users/*/Downloads/omnetpp-6.1-windows-x86_64/omnetpp-6.1/samples/DcwTest/src'
make: *** [Makefile:2: all] Error 2**
"make MODE=release all" terminated with exit code 2. Build might be incomplete."
```

![Знімок екрана 2024-11-21 001604](https://github.com/user-attachments/assets/106dace5-e7d8-49da-97a8-bbb0ffa039ad)

**Comment by Abh4git** [comment]
Hope you could overcome this. Most probably the relative paths are not as expected. (INET not in expected path possibly)
